source/utils/types.ts

```diff
@@ -50,19 +51,34 @@ export interface InterfaceWithDefaults extends Instance {
 	};
 }
 
-interface RetryOption {
-	retries?: ((retry: number, error: Error) => number) | number;
+export type RetryFn = (retry: number, error: Error) => number;
+
+export interface RetryOption {
+	retries?: RetryFn | number;
 	methods?: Method[];
 	statusCodes?: StatusCode[];
 	maxRetryAfter?: number;
 	errorCodes?: ErrorCode[];
 }
 
+export type RequestFn = typeof https.request;
+
 export interface MergedOptions extends Options {
 	retry: RetryOption;
 }
 
-export interface Options extends RequestOptions {
+export interface Delays {
+	lookup?: number;
+	connect?: number;
+	secureConnect?: number;
+	socket?: number;
+	response?: number;
+	send?: number;
+	request?: number;
+}
+
+// The library overrides the type definition of `agent`.
+export interface Options extends Pick<https.RequestOptions, Exclude<keyof https.RequestOptions, 'agent'>> {
```

Comment:
`Options` is a particularly tricky case.

One challenge is that got supports options which have slightly different types than `https.RequestOptions`, which causes it to be type-wise incompatible. Using Pick/Exclude here seems like a partial solution for the first issue, though it requires coercion back to `https.RequestOptions` in `request-as-event-emitter.ts`.

Another is that `preNormalizeArguments` and `normalizeOptions` narrow down the option types. I haven't addressed that yet.

Reply:
Would be better to use `Omit` from https://github.com/sindresorhus/type-fest#utilities